test/framework/src/main/java/org/elasticsearch/cluster/coordination/DeterministicTaskQueue.java

```diff
@@ -323,12 +326,13 @@ public ThreadPoolStats stats() {
 
             @Override
             public ExecutorService generic() {
-                return getExecutorService(runnableWrapper);
+                return executor(Names.GENERIC);
             }
 
             @Override
             public ExecutorService executor(String name) {
-                return getExecutorService(runnableWrapper);
+                return executors.computeIfAbsent(name, key -> getExecutorService(runnableWrapper,
+                        Names.SAME.equals(key) ? Runnable::run : DeterministicTaskQueue.this::scheduleNow));
```

Comment:
Could we instead return `EsExecutors.newDirectExecutorService();` if `Names.SAME.equals(name)`?

Should we also return the same (non-direct) executor service each time too rather than constructing a new one each time?

Reply:
> Should we also return the same (non-direct) executor service each time too rather than constructing a new one each time?

I mean for all other names, your change to cache them by name looks good but I don't think we need to distinguish the non-direct executors from each other either.

Reply:
Yea you're right on both points. Much simpler now :)